java/client/src/main/java/glide/api/commands/StreamBaseCommands.java

```diff
@@ -181,4 +181,82 @@ public interface StreamBaseCommands {
      */
     CompletableFuture<Map<String, String[]>> xrange(
             String key, StreamRange start, StreamRange end, long count);
+
+    /**
+     * Returns stream entries matching a given range of IDs in reverse order.<br>
+     * Equivalent to {@link #xrange(String, StreamRange, StreamRange)} but returns the entries in
+     * reverse order.
+     *
+     * @param key The key of the stream.
+     * @param end Ending stream ID bound for range.
+     *     <ul>
+     *       <li>Use {@link IdBound#of} to specify a stream ID.
+     *       <li>Use {@link IdBound#ofExclusive} to specify an exclusive bounded stream ID.
+     *       <li>Use {@link InfRangeBound#MAX} to end with the maximum available ID.
+     *     </ul>
+     *
+     * @param start Starting stream ID bound for range.
+     *     <ul>
+     *       <li>Use {@link IdBound#of} to specify a stream ID.
+     *       <li>Use {@link IdBound#ofExclusive} to specify an exclusive bounded stream ID.
+     *       <li>Use {@link InfRangeBound#MIN} to start with the minimum available ID.
+     *     </ul>
+     *
+     * @return A <code>Map</code> of key to stream entry data, where entry data is an array of item
+     *     pairings.
+     * @example
+     *     <pre>{@code
+     * // Retrieve all stream entries
+     * Map<String, String[]> result = client.xrevrange("key", InfRangeBound.MAX, InfRangeBound.MIN).get();
+     * result.forEach((k, v) -> {
+     *     System.out.println("Stream ID: " + k);
+     *     for (int i = 0; i < v.length;) {
+     *         System.out.println(v[i++] + ": " + v[i++]);
+     *     }
+     * });
+     * // Retrieve exactly one stream entry by id
+     * Map<String, String[]> result = client.xrevrange("key", IdBound.of(streamId), IdBound.of(streamId)).get();
+     * System.out.println("Stream ID: " + streamid + " -> " + Arrays.toString(result.get(streamid)));
+     * }</pre>
+     */
+    CompletableFuture<Map<String, String[]>> xrevrange(
```

Comment:
Should be `String[][]` to match `xread`
```suggestion
    CompletableFuture<Map<String, String[][]>> xrevrange(
```

Reply:
Sorry, please update docs and examples and transaction and tests too...

Reply:
I will have to update xrange too